ibis/spark/udf.py

```diff
@@ -32,9 +32,10 @@ def validate_func_and_types(self, func):
         if not callable(func):
             raise TypeError('func must be callable, got {}'.format(func))
 
-        # validate that the input_type argument and the function signature
-        # match
-        valid_function_signature(self.input_type, func)
+        # Validate that the input_type argument and the function signature
+        # match and that the output_type is valid
+        v.validate_input_type_count(self.input_type, func)
```

Comment:
Don't need to touch Spark udf either. Only PySpark and Pandas backend supports "vectorized UDF".